sw/device/lib/dif/autogen/dif_adc_ctrl_autogen_unittest.cc

```diff
@@ -150,6 +150,34 @@ TEST_F(IrqIsPendingTest, Success) {
   EXPECT_TRUE(irq_state);
 }
 
+class AcknowledgeStateTest : public AdcCtrlTest {};
+
+TEST_F(AcknowledgeStateTest, NullArgs) {
+  dif_adc_ctrl_irq_state_snapshot_t irq_snapshot = 0;
+  EXPECT_DIF_BADARG(dif_adc_ctrl_irq_acknowledge_state(nullptr, irq_snapshot));
+}
+
+TEST_F(AcknowledgeStateTest, AckSnapshot) {
+  const uint32_t num_irqs = 1;
+  dif_adc_ctrl_irq_state_snapshot_t irq_snapshot = 1;
+
+  // Test a few snapshots.
+  for (size_t i = 0; i < num_irqs; ++i) {
+    irq_snapshot = ~irq_snapshot & (num_irqs - 1);
```

Comment:
Oops, meant this to be...

```suggestion
    irq_snapshot = ~irq_snapshot & ((1u << num_irqs) - 1);
```